src/player.js

```diff
@@ -1224,11 +1241,25 @@ export default class Player extends FakeEventTarget {
   _setDefaultTracks(): void {
     const activeTracks = this.getActiveTracks();
     const playbackConfig = this._config.playback;
+    const configTextLang = playbackConfig.textLanguage;
+    const fallbackTextTrack: ?Track =  this._getTracksByType(TrackTypes.TEXT).find(track => TextTrack.langComparer(OFF, track.language));
+    let textLanguage = '';
 
     this.hideTextTrack();
 
-    const textLanguage = (playbackConfig.textLanguage === "auto") ? Locale.language : playbackConfig.textLanguage;
-    this._setDefaultTrack(TrackTypes.TEXT, textLanguage, activeTracks.text);
+    if (configTextLang === AUTO) {
+      if (activeTracks.text) {
+        textLanguage = activeTracks.text.language;
+      } else {
+        textLanguage = Locale.language;
+      }
+    } else if (configTextLang !== '') {
```

Comment:
and you'll not need this condition
if textLanguage is not auto, it will be the explicit lang if user defined this or empty (because of the player default config) if not.